cmd/argocd/commands/app.go

```diff
@@ -1603,15 +1606,16 @@ func groupResourceStates(app *argoappv1.Application, selectedResources []argoapp
 	return resStates
 }
 
-func checkResourceStatus(watchSync bool, watchHealth bool, watchOperation bool, watchSuspended bool, healthStatus string, syncStatus string, operationStatus *argoappv1.Operation) bool {
+func checkResourceStatus(watchSync bool, watchHealth bool, watchOperation bool, watchSuspended bool, watchDegraded bool, healthStatus string, syncStatus string, operationStatus *argoappv1.Operation) bool {
 	healthCheckPassed := true
-	if watchSuspended && watchHealth {
```

Comment:
Passing only the `--healthy` flag would wait until the resource is healthy. Shouldn't the condition for multiple health checks be `if watchSuspended && watchHealth && watchDegraded {` ?

Reply:
@chetan-rns So, for this one, I'm not too sure. I wanted to handle the following combinations:
* `--health` only
* `--degraded` only
* `--suspended` only
* `--health --suspended`
* `--health --degraded`
* `--health --suspended --degraded`

I wasn't too sure what would be the best way to structure the code for that, and I went for having this `if watchHealth` that sets `healthCheckPassed` using some `or` conditions. This should keep the previous flag combinations that worked, while adding new combinations with the `--degraded` flag.

I think this whole if/else if/else if could also be simplified to:

```
if watchSuspended || watchHealth || watchDegraded {
		healthCheckPassed = (watchHealth && healthStatus == string(health.HealthStatusHealthy)) ||
			(watchSuspended && healthStatus == string(health.HealthStatusSuspended)) ||
			(watchDegraded && healthStatus == string(health.HealthStatusDegraded))
}
```

If feel if we use some `and` conditions (`if watchSuspended && watchHealth && watchDegraded {`), we will need more if/else if branches to make these combinations work.

Do you have a preference on which approach I should use?